---
title: "Synthesis Exercise: Microarray Data Analysis"
author: "Guillem Cortiada Rovira"
date: "4/6/2020"
e-mail: guillem.cortiada@alum.esci.upf.edu
GitHub: <https://github.com/guillem99/Omics_techniques_2020.git>
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(Biobase)
library(BiocManager)
library(ggplot2)
library(GEOquery)
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("mogene10sttranscriptcluster.db")
```

## Introduction
All codes will be available in my GitHub: https://github.com/guillem99/Omics_techniques_2020.git 

## Title
Topical protein tyrosine kinase inhibitor effect on graft model of alopecia areata: time course

# Description of the Study
# Summary
Analysis of skin from C3H/HeJ grafted recipients with established alopecia areata (AA) following alopecic graft transplantation that were treated with topical Janus kinase (JAK) inhibitor ruxolitinib or tofacitinib for up to 24 wks. Results provide insight into molecular basis of AA pathogenesis.

# Objective
The aim of the study is the genes expressions patterns after an alopecic graft transplant.

## Information on the Experimental Design
# Experiment Design
We have three groups of mouse that after an alopecic graft transplantation skin his skin were taken at initial time, 6, 12 and 24 weeks (in 24 weeks, we only have taken the 2 groups with treatment and only two samples in each one) after administration of ruxolitinib(jak1i), tofacitinib(jak3i) or reference substance role.
It is a group design (Expression profiling by array)

# Response Variable
Gene expression patterns

# Groups
3 groups (reference, treated with tofacitinib(jak3i), treated with ruxolitinib(jak1i)) (3 biological replicates were taken at each time point for each treatment modality)

·Sample Size = 31 samples:
  3 assays: initial time point, reference substance role
	3 assays: initial time point, tofacitinib
	3 assays: initial time point, ruxolitinib
	3 assays: 6 weeks, reference substance role
	3 assays: 6 weeks, tofacitinib
	3 assays: 6 weeks, ruxolitinib
	3 assays: 12 weeks, reference substance role
	3 assays: 12 weeks, tofacitinib
	3 assays: 12 weeks, ruxolitinib
	2 assays: 24 weeks, tofacitinib
	2 assays: 24 weeks, ruxolitinib
- In 0, 6 and 12 weeks = (3(reference)+3(tofacitinib)+3(ruxolitinib))*3 = 27
- In 24 weeks only 2(tofacitinib)+2(ruxolitinib)

·Experimental Factors:
	Batch = (Block) Scan date
	Treatment = (treatment) Vehicle (PBS), tofacitinib, ruxolitinib
	Timepoint = (timepoint) 0 weeks, 6 weeks, 12 weeks, 24 weeks

# Organism
Mus Musculus

# Platform
GPL1261: Affymetrix Mouse Genome 430 2.0 Array

# Citations
Xing L, Dai Z, Jabbari A, Cerise JE et al. Alopecia areata is driven by cytotoxic T lymphocytes and is reversed by JAK inhibition. Nat Med 2014 Sep;20(9):1043-9. PMID: 25129481

# GEO Dataset

```{r}
gse <- getGEO("GSE45514", GSEMatrix = TRUE)
summary(gse)
```

# Description of the data

```{r}
show(gse)
class(gse)
names(gse)
eset <- gse[[1]]
print("esetFromGeo")
eset
targets <- pData(eset)
print("Targets")
head(targets)
```

Maximum, minimum and mean for each group (Jak1i, Jak3i and PBS)

```{r}
eset_jak1i <- exprs(eset[,1:11])
eset_jak3i <- exprs(eset[,12:22])
eset_pbs <- exprs(eset[,23:31])

# Jak1i
print(paste("Minimum = ", min(eset_jak1i) ))
print(paste("Maximum = ", max(eset_jak1i) ))
print(paste("Mean = ", mean(eset_jak1i)))

# Jak3i
print(paste("Minimum = ", min(eset_jak3i)))
print(paste("Maximum = ", max(eset_jak3i)))
print(paste("Mean = ", mean(eset_jak3i)))

# PBS
print(paste("Minimum = ", min(eset_pbs)))
print(paste("Maximum = ", max(eset_pbs)))
print(paste("Mean = ", mean(eset_pbs)))
```


```{r}
print("ExpressionSet")
gene_expresion <- exprs(eset)

print(paste("Number of genes and number of samples = ", dim(gene_expresion)))

summary(gene_expresion)
hist(gene_expresion)
print("As we can observe, the data needs to be normalized (applying logaritms)")
```


Normalizing the data

```{r}
hist(log(gene_expresion))
print("The data looks better. So, we need to apply this transformation to the original data.")
exprs(eset) <- log(exprs(eset))
```


Boxplots

```{r}
pcs <- prcomp(exprs(eset))
names(pcs)
boxplot(pcs$sdev)
plot(pcs$rotation[,1], pcs$rotation[,2], main="Representation of first two principal components")
text(pcs$rotation[,1], pcs$rotation[,2])
```


```{r}
print("To see the top variable genes")
genes <- rownames(gene_expresion)
variab <- apply(gene_expresion,1, sd)
orderedGenes <- genes[order(variab, decreasing=TRUE)]
head(variab[order(variab, decreasing=TRUE)])
```

Analyse the Top genes applying Barplots

```{r}
# First Top gene
barplot(gene_expresion["1427801_at",])

# Second Top gene
barplot(gene_expresion["1425431_at",])

# Third Top gene
barplot(gene_expresion["1425655_at",])
```


## Comparisions to be performed between groups
In our study, we have three groups after an alopecic graft transplantation, skin with treatment of ruxolitinib(jak1i), with treatment of tofacitinib(jak3i) or with a reference substance.
Hence, our comparisions should be to compare group of jak1i to group jak3i and reference substance and jak3i to the reference substance.
Question: How does affect the treatments after an alopecic graft transplant?


## Build the Design Matrix


```{r design-Matrix}
library(limma)
design_matrix <- matrix(
  c(1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
    0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,
    0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1), nrow=31)
design2 <- model.matrix(~0+targets$`treatment:ch1`)
design2
colnames(design_matrix)<- colnames(design2)<-c("jak1i", "jak3i", "PBS")
rownames(design_matrix) <- rownames(design2)<- targets$source_name_ch1

print(design_matrix); print(design2)
```

## Contrast Matrix

```{r contrast-Matrix}
contrast_matrix <- makeContrasts (
  Jak1ivsJak3i = jak3i-jak1i,
  Jak1ivsPBS = PBS-jak1i,
  Jak3ivsPBS = PBS-jak3i,
  levels=design_matrix)
contrast_matrix
```


# Diferential expressed genes selection
# Fit a linear Model

```{r}
dataMatrix <- gse[[1]]
colnames(dataMatrix)==rownames(targets)
```


```{r fitModel}
fit <- lmFit(dataMatrix, design_matrix)
fit_main <- contrasts.fit(fit, contrast_matrix)
fit_main <- eBayes(fit_main)
```

# Estimate Contrasts "TopTable"

```{r extractResults}
## Jak1i vs Jak3i
topTab_Jak1ivsJak3i <- topTable (fit_main, number=nrow(fit_main), coef="Jak1ivsJak3i", adjust="fdr"); head(topTab_Jak1ivsJak3i)

## Jak1i vs PBS
topTab_Jak1ivsPBS <- topTable (fit_main, number=nrow(fit_main), coef="Jak1ivsPBS", adjust="fdr"); head(topTab_Jak1ivsPBS)

## Jak3i vs PBS
topTab_Jak3ivsPBS  <- topTable (fit_main, number=nrow(fit_main) , coef="Jak3ivsPBS", adjust="fdr"); head(topTab_Jak3ivsPBS)
```

# Volcano Plot

```{r showResults}
## Jak1i vs Jak3i
volcanoplot(fit_main, coef="Jak1ivsJak3i", highlight=10)

## Jak1i vs PBS
volcanoplot(fit_main, coef="Jak1ivsPBS", highlight=10)

## Jak3i vs PBS
volcanoplot(fit_main, coef="Jak3ivsPBS", highlight=10)
```


# Convert our probest identifiers into gene symbols

```{r}
library(mogene10sttranscriptcluster.db)
all_anota <- data.frame(exprs(eset))
Annot <- data.frame(SYMBOL=sapply(contents(mogene10sttranscriptclusterSYMBOL), paste, collapse=", "),
                    DESC=sapply(contents(mogene10sttranscriptclusterGENENAME), paste, collapse=", "))
Annot<-Annot[!Annot$SYMBOL=="NA",]
Annot<-Annot[!Annot$DESC=="NA",]
head(Annot)

anotaGenes <- merge(Annot, all_anota, by.x=0,by.y=0)
head(anotaGenes)
write.table(anotaGenes,file="data.ann.txt",sep="\t")

rownames(anotaGenes)<-anotaGenes[,1]
anotaGenes <- anotaGenes[,-1]

anotaGenes.end <- merge(anotaGenes,topTab_Jak1ivsJak3i, by.x=0,by.y=0)
topTab.end <- anotaGenes.end[,c(1:3,12:17,4:11)]
topTab.end <- topTab.end[order(topTab.end$SYMBOL),]
rownames(topTab.end)<-topTab.end[,1]
write.csv(topTab.end,file="TopTable.end.csv")
```





